File: problem_003.go
// -*- compile-command: "go build -o problem_003_go problem_003.go" -*-
// Copyright (c) 2017 Michael Caldwell
package main

import "fmt"

func main() {
        fmt.Println("Project Euler - Problem 3:")
        fmt.Println("What is the largest prime factor of the number 600851475143?\n")

        n := 600851475143
	factor := 2

        for factor < n {
                for n % factor == 0 {
                         n /= factor
                }
		factor++
        }

        fmt.Println(n)
}

File: problem_003.kt
// -*- compile-command: "kotlinc problem_003.kt -include-runtime -d problem_003.jar" -*-
// Copyright (c) 2017 Michael Caldwell

fun main(args: Array<String>) {
        println("Project Euler - Problem 3:")
        println("What is the largest prime factor of the number 600851475143?\n")

        var n: Long = 600851475143
        var d: Long = 1

        while (n > 1) {
                d += 1
                while (n % d == 0.toLong()) {
                        n /= d
                }
        }

        println(d)
}
